std/datetime/systime.d

```diff
@@ -8520,19 +8519,18 @@ public:
         if (isSomeString!(S))
     {
         import std.algorithm.searching : countUntil, find;
-        import std.conv : to;
         import std.string : strip;
 
-        auto dstr = to!dstring(strip(isoExtString));
+        auto str = strip(isoExtString);
 
-        auto tIndex = dstr.countUntil('T');
+        auto tIndex = str.countUntil('T');
         enforce(tIndex != -1, new DateTimeException(format("Invalid ISO Extended String: %s", isoExtString)));
 
-        auto found = dstr[tIndex + 1 .. $].find('.', 'Z', '+', '-');
-        auto dateTimeStr = dstr[0 .. $ - found[0].length];
+        auto found = str[tIndex + 1 .. $].find('.', 'Z', '+', '-');
```

Comment:
`countUntil` still works in terms of code points, no? `tIndex` will not be the proper index when there are non-ASCII characters before the 'T'.